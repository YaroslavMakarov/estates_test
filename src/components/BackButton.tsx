import React from 'react';
import { Container, Button } from 'react-bootstrap';
import 'bootstrap/dist/css/bootstrap.min.css';
import { useHistory } from 'react-router-dom';

const BackButton = () => {
    const history = useHistory();

    return (
        <Container
            fluid
            className="p-0"
        >
            <Button
                variant="dark"
                className="mb-3"
                onClick={() => history.goBack()}
            >
                Back
            </Button>
        </Container>
    );
};

export default BackButton;