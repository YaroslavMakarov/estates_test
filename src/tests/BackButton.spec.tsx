import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

import BackButton from '../components/BackButton';

describe('BackButton component', () => {
    const wrapper = mount(<MemoryRouter><BackButton /></MemoryRouter>);

    it('get BackButton snapShot', () => {
        expect(wrapper.debug()).toMatchSnapshot();
    })
});