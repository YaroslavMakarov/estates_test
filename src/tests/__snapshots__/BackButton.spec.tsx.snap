// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton component get BackButton snapShot 1`] = `
"<MemoryRouter>
  <Router history={{...}}>
    <BackButton>
      <Container fluid={true} className=\\"p-0\\">
        <div className=\\"p-0 container-fluid\\">
          <Button variant=\\"dark\\" className=\\"mb-3\\" onClick={[Function: onClick]} active={false} disabled={false}>
            <button onClick={[Function: onClick]} disabled={false} type=\\"button\\" className=\\"mb-3 btn btn-dark\\">
              Back
            </button>
          </Button>
        </div>
      </Container>
    </BackButton>
  </Router>
</MemoryRouter>"
`;
